src/main/java/io/gingersnapproject/cdc/configuration/DefaultRuleParser.java

```diff
@@ -0,0 +1,148 @@
+package io.gingersnapproject.cdc.configuration;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.infinispan.commons.dataconversion.internal.Json;
+
+public final class DefaultRuleParser {
+
+   private DefaultRuleParser() { }
+
+   public static Rule.SingleRule parse(String body) {
+      Json json = Json.read(body);
+      return new SingleRuleConcrete(json);
+   }
+
+   private static class SingleRuleConcrete implements Rule.SingleRule {
+      private final Connector connector;
+      private final Database database;
+      private final Backend backend;
```

Comment:
Some of the info here comes from the Cache CR. This could complicate the Informer implementation. see #42